components/script/dom/mutationobserver.rs

```diff
@@ -37,4 +72,154 @@ impl MutationObserver {
         ScriptThread::add_mutation_observer(&*observer);
         Ok(observer)
     }
+
+    /// https://dom.spec.whatwg.org/#queue-a-mutation-observer-compound-microtask
+    /// Queue a Mutation Observer compound Microtask.
+    pub fn queueMutationObserverCompoundMicrotask(compoundMicrotask: Microtask) {
+        // Step 1
+        if ScriptThread::is_mutation_observer_compound_microtask_queued() {
+            return;
+        }
+        // Step 2
+        ScriptThread::set_mutation_observer_compound_microtask_queued(true);
+        // Step 3
+        ScriptThread::enqueue_microtask(compoundMicrotask);
+    }
+
+    /// https://dom.spec.whatwg.org/#notify-mutation-observers
+    /// Notify Mutation Observers.
+    pub fn notifyMutationObservers() {
+        // Step 1
+        ScriptThread::set_mutation_observer_compound_microtask_queued(false);
+        // Step 2
+        let notifyList = ScriptThread::get_mutation_observer();
+        // Step 3, Step 4 not needed as Servo doesn't implement anything related to slots yet.
+        // Step 5: Ignore the specific text about execute a compound microtask.
```

Comment:
Sorry, I think I said something that was misinterpreted. We definitely need step 5 or nothing actually happens here and the mutation observers' callbacks are never invoked. Add a TODO for step 5.3, but the rest are necessary.